tests/test_http_parser.py

```diff
@@ -730,6 +730,21 @@ def test_http_response_parser_no_reason(response: Any) -> None:
     assert msg.reason == ""
 
 
+def test_http_response_parser_lenient_headers(response: Any) -> None:
+    messages, upgrade, tail = response.feed_data(
+        b"HTTP/1.1 200 test\r\nFoo: abc\x01def\r\n\r\n"
+    )
+    msg = messages[0][0]
+
+    assert msg.headers["Foo"] == "abc\x01def"
+
+
+@pytest.mark.dev_mode
+def test_http_response_parser_strict_headers(response: Any) -> None:
+    with pytest.raises(http_exceptions.BadHttpMessage):
```

Comment:
It's highly recommended to always use the `match=` argument with `pytest.raises()`. Use a full regex, starting with `^` and ending with `$`.

Reply:
Yep, but for the parser this seems too complex, as the messages might not be the same between py/c parsers. Plus, the contents of the message from llhttp is not our concern, we might end up needing to update the regex with every version of llhttp (I'm sure they have their own tests for error messages).

Reply:
But if this starts erroring out for a different reason, our test would still fail. Shouldn't we normalize those, then? Besides, since we always update llhttp to a pinned version, it's okay to update the error message.

Reply:
Well, BadHttpMessage is a parse error. I don't really care what it thinks the error is beyond that, I just care that it thinks this is bad in dev mode, but not normally.

Normalising the exception messages and keeping them in sync seems like a big job to me, and not one I currently want to take on.

> Besides, since we always update llhttp to a pinned version, it's okay to update the error message.

My concern is the maintenance burden of updating llhttp. I'm not going to be happy if I have to manually update 106 tests (so far), just to bump the version...

Reply:
okay, let's keep it as is. resolving